src/Equinox.Core/CachingDecorator.fs

```diff
@@ -0,0 +1,51 @@
+namespace Equinox.Core
+
+open Serilog
+open System
+open System.Threading
+open System.Threading.Tasks
+
+type IReloadableCategory<'event, 'state, 'context> =
+    inherit ICategory<'event, 'state, 'context>
+
+    abstract Reload: log: ILogger * streamName: string * requireLeader: bool * streamToken: StreamToken * state: 'state * ct: CancellationToken
```

Comment:
Aren't all categories reloadable? can we avoid adding another interface here? Reloading could default to returning the current token and state, or the empty token and state

Reply:
Or is the point more that the stores should implement reloadable categories, while consumers of the library deal with ICategories?

Reply:
Alright, read through the rest of the code, looks like all the stores primarily implement this concept rather than the ICategory directly. I wonder if we should make this definition stand on its own rather than inherit from ICategory since users will never see this type anyway and inheriting, while powerful, gives me the shivers

Reply:
Good point that I'll take away and ponder...

You are correct that MemoryStore could reasonably implement it too, and it's a pretty central part of any reasonably efficient even sourced infra (i.e. a reload of kinds needs to happen in the case of a resync even if you are not doing caching), and/or provide a null impl
A big reason that the file is not compiled into Core is that this is just an impl detail.
It may also be possible and/or reasonable to twist things such that the CachingStrategy DU resides in Core and the impls don't need the duplicated mapping/wiring.

I had been resisting having a big Core lib from the very start - being able to point people to 2/3 files: https://github.com/jet/equinox/blob/master/src/Equinox/Core.fs + https://github.com/jet/equinox/blob/master/src/Equinox.MemoryStore/MemoryStore.fs and saying "that's all it is" is a key value prop

Stuff like 
- the prohibition on caching with EventsAreState Mode https://github.com/jet/equinox/blob/doc-tweaks/src/Equinox.EventStoreDb/EventStoreDb.fs#L446-L451 (I couldn't think of a cleaner way to wire it at the time - it's not a critical design masterplan)
- the mandatory/explicit caching mode for Cosmos/Dynamo (having to say NoCaching vs not specifying a Caching in the other stores)
becomes more debatable if making specifying the caching can simply be made orthogonal

Reply:
Merging...
- MemoryStore is not currently reloadable - still on the fence about whether it's important that the memorystore be minimal and demonstrates caching is optional
- open to a PR to make `IReloadable` a separate interface not derived from `ICategory`
- ICategory is prob a bad name